grid-client/deployer/k8s_deployer.go

```diff
@@ -51,7 +51,11 @@ func (d *K8sDeployer) Validate(ctx context.Context, k8sCluster *workloads.K8sClu
 
```

Comment:
this file changes is risky (I didn't complete them but they need more focus I guess)